### PR TITLE
Hydrogen-brine simulations - H2STORE 

### PR DESCRIPTION
**Pull request by svenn-t**
This PR adds hydrogen-brine fluid system via a new H2STORE keyword. The approach is exactly the same as in CO2STORE, so it works with OIL/GAS and WATER/GAS systems, as well as DISGAS/DISGASW and DIFFUSE. Since I couldn't find information on thermal properties of dissolved hydrogen in brine, it should not be used with the THERMAL keyword. 

Dissolved hydrogen in brine was modeled with [1]. Hydrogen gas density was modeled with [2]. Other properties are modeled the same way as in CO2STORE but modified for hydrogen-brine (similar approach has been done recently [3]).

[1] [https://doi.org/10.1016/j.ijhydene.2017.07.228](https://doi.org/10.1016/j.ijhydene.2017.07.228)
[2] [https://doi.org/10.1063/1.3160306](https://doi.org/10.1063/1.3160306)
[3] [https://doi.org/10.1016/j.ijhydene.2022.09.222](https://doi.org/10.1016/j.ijhydene.2022.09.222)

### CODE REVIEW

File: tests/parser/RunspecTests.cpp [hunk 1]
```diff
@@ -434,55 +434,6 @@ WELLDIMS
     BOOST_CHECK_EQUAL(wd.maxGroupsInField(), 0);  // WELLDIMS(3), defaulted
 }
 
-BOOST_AUTO_TEST_CASE(WELLDIMS_MaxWList_Dflt)
-{
```
[the remark this answers is not in the document]
Comment (svenn-t) [reply tests/parser/RunspecTests.cpp]: Sorry, definitely a mistake/rebase mishap! I re-added the deleted code section.

**Comment by totto82** [comment]
jenkins build this please

**Comment by svenn-t** [comment]
> I haven't looked at this code at all, but I notice that there are a number of [new warnings](https://ci.opm-project.org/job/opm-common-PR-builder/5612/gcc/new/) emitted as a result of this work. Some of the new warnings are **mostly** benign (e.g., unused parameters), but the narrowing conversions and missing `return` statements in functions not returning `void` must be addressed before this is ready for merging. Ideally we will address every warning in the code we control.

I've updated the code. That should address the narrowing, unused-parameter and return-type warnings (they were eliminated when I compiled it at least).

**Comment by bska** [comment]
> > I haven't looked at this code at all, but I notice that there are a number of [new warnings](https://ci.opm-project.org/job/opm-common-PR-builder/5612/gcc/new/) emitted as a result of this work.
> 
> I've updated the code. That should address the narrowing, unused-parameter and return-type warnings (they were eliminated when I compiled it at least).

Thanks a lot for the quick response.  I'll run another build check to get the CI system's assessment of the updated code.

**Comment by bska** [comment]
jenkins build this please

**Comment by bska** [comment]
> > I've updated the code. That should address the narrowing, unused-parameter and return-type warnings (they were eliminated when I compiled it at least).
> 
> Thanks a lot for the quick response. I'll run another build check to get the CI system's assessment of the updated code.

The new version of this PR does indeed [fix those issues](https://ci.opm-project.org/job/opm-common-PR-builder/5613/gcc/fixed/).  Thanks a lot for the update.

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
Thanks for the contribution. I have looked through the code I from a coding perspective this is ok. I follows the CO2STORE implementation. I also think it is a valuable feature to have in Flow and it is completely orthogonal to the rest of the PVT implementation in Flow so I suggest we merge this before the release? Any other thoughts?

**Comment by totto82** [comment]
jenkins build this please

**Comment by svenn-t** [comment]
I didn't quite understand the jenkins build errors (if there were any?), but I rebased the code just in case. Hopefully there will be no build errors now. (Sorry if this is not the way I'm suppose to do rebasing in PRs!)

**Comment by svenn-t** [comment]
Hopefully now the build should go through. Sorry about that!

**Comment by totto82** [comment]
jenkins build this please

**Comment by totto82** [comment]
jenkins build this please

**Pull request merged**